### ISSUE TITLE
`singlejar` overwrites files in` META-INF/proguard` dirs

### ISSUE THREAD

**Issue by snazhmudinov** [bazelbuild/bazel#20277]
### Description of the bug:

`singlejar` tool overwrites the content of `META-INF/proguard` folder, if there are multiples files with the same relative paths.

### Which category does this issue belong to?

Java Rules

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Create 2 simple jars. Each containing the `META-INF/proguard/rules.pro`. The content of rules.pro files should be different. 
Use those jar dependencies in `java_binary` and build `*_deploy.jar` target. 
Unzip the output jar, and check the content of `META-INF/proguard/rules.pro`.

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

6.3.1

### Have you found anything relevant by searching the web?

Asked in bazel slack, there was no response:
https://bazelbuild.slack.com/archives/CA31HN1T3/p1700243516385089

**Comment by hvadehra** [comment]
cc @cushon

**Comment by snazhmudinov** [comment]
Might be similar: https://github.com/bazelbuild/bazel/issues/7330

**Comment by hvadehra** [comment]
Yeah, I agree this is a more narrowly scoped version of #7330. I think we should use that to track this.

**Comment by snazhmudinov** [comment]
@hvadehra thanks! I see the prio of the linked issue is set to 2. Do you know if that something we should expect to be fixed soon?

**Comment by hvadehra** [comment]
Yes, I will look into this soon (bumped from p3 to p2 to reflect that).
